### PR TITLE
FIX: show hide bootstrap mode notice in real time

### PR DESCRIPTION
**Pull request by techAPJ**

### CODE REVIEW

**Comment by techAPJ** [comment]
Test failure looks unrelated.

**Review by ZogStriP — COMMENTED** [review]
Any chances you could add a test?

**Comment by techAPJ** [comment]
We already have a test for bootstrap mode notice [here](https://github.com/discourse/discourse/blob/42c64abe156829cc83b677e7a46d481aa97c380e/app/assets/javascripts/discourse/tests/acceptance/bootstrap-mode-notice-test.js).

This commit fixes a very edge case where if the site setting `bootstrap_mode_enabled` is disabled in a Sidekiq job than the banner should be hidden instantaneously on the client side. I am not sure if we can repro this particular case in a test because it requires real time changes on both server side and client side.

**Comment by ZogStriP** [comment]
> This commit fixes a very edge case where if the site setting `bootstrap_mode_enabled` is disabled in a Sidekiq job than the banner should be hidden instantaneously on the client side. I am not sure if we can repro this particular case in a test because it requires real time changes on both server side and client side.

You don't need to test all that. Just that when you change the `bootstrap_mode_enabled` site setting, the bootstrap notice is gone.

File: app/assets/javascripts/discourse/tests/acceptance/bootstrap-mode-notice-test.js [hunk 1]
```diff
@@ -34,5 +35,11 @@ acceptance("Bootstrap Mode Notice", function (needs) {
       "/wizard/steps/hello-world",
       "it transitions to the wizard page"
     );
+
+    set(this.siteSettings, "bootstrap_mode_enabled", false)
+    assert.ok(
+      !exists(".bootstrap-mode-notice"),
+      "does not has the bootstrap mode notice"
```
Comment (ZogStriP) [inline app/assets/javascripts/discourse/tests/acceptance/bootstrap-mode-notice-test.js:42]: Let's try to be more descriptive in the test description

```suggestion
      "removes the bootstrap mode notice when bootstrap mode is disabled"
```

Comment (techAPJ) [reply app/assets/javascripts/discourse/tests/acceptance/bootstrap-mode-notice-test.js:42]: sure, done.

**Review by ZogStriP — APPROVED** [review]

File: app/assets/javascripts/discourse/tests/acceptance/bootstrap-mode-notice-test.js [hunk 2]
```diff
@@ -34,5 +35,12 @@ acceptance("Bootstrap Mode Notice", function (needs) {
       "/wizard/steps/hello-world",
       "it transitions to the wizard page"
     );
+
+    set(this.siteSettings, "bootstrap_mode_enabled", false);
+    await settled();
+    assert.ok(
+      !exists(".bootstrap-mode-notice"),
+      "removes the bootstrap mode notice when bootstrap mode is disabled"
```
Comment (ZogStriP) [inline app/assets/javascripts/discourse/tests/acceptance/bootstrap-mode-notice-test.js:43]: We could be less verbose (sorry 😅)

```suggestion
      "removes the notice when bootstrap mode is disabled"
```

File: app/assets/javascripts/discourse/tests/acceptance/bootstrap-mode-notice-test.js [hunk 3]
```diff
@@ -34,5 +35,12 @@ acceptance("Bootstrap Mode Notice", function (needs) {
       "/wizard/steps/hello-world",
       "it transitions to the wizard page"
     );
+
+    set(this.siteSettings, "bootstrap_mode_enabled", false);
+    await settled();
+    assert.ok(
```
Comment (davidtaylorhq) [inline app/assets/javascripts/discourse/tests/acceptance/bootstrap-mode-notice-test.js:41]: We're on the Wizard page at this point, right? So we would never expect the banner to show? I think we need to be on the homepage for this test to be meaningful?

```suggestion
    visit("/");
    set(this.siteSettings, "bootstrap_mode_enabled", false);
    await settled();
    assert.ok(
```

Comment (techAPJ) [reply app/assets/javascripts/discourse/tests/acceptance/bootstrap-mode-notice-test.js:42]: That is absolutely correct! Pushed a fix. 👍🏼

Comment (ZogStriP) [reply app/assets/javascripts/discourse/tests/acceptance/bootstrap-mode-notice-test.js:42]: Don't you need to `await` the `visit("/")`?

Comment (techAPJ) [reply app/assets/javascripts/discourse/tests/acceptance/bootstrap-mode-notice-test.js:42]: @davidtaylorhq does the `await settled()` takes care of `visit("/")` or should we use `await visit("/")` explicitly here?

Comment (davidtaylorhq) [reply app/assets/javascripts/discourse/tests/acceptance/bootstrap-mode-notice-test.js:42]: We should `await visit()` as well - my bad 👍

**Review by davidtaylorhq — APPROVED** [review]
